catalog/transformer/catalog-transformer-pptx/src/test/java/ddf/catalog/transformer/input/pptx/PptxInputTransformerTest.java

```diff
@@ -0,0 +1,224 @@
+    public void testTransformNullInput() throws IOException, CatalogTransformerException {
+        PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+        t.transform(null);
+    }
+
+    @Test
+    public void testBunchOfPptxFiles() throws IOException, CatalogTransformerException {
+        for (String file : files) {
+            PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+            try (InputStream is = getResource(file)) {
+                Metacard mc = t.transform(is);
+                assertNotNull(mc.getThumbnail());
+            }
+        }
+    }
+
+    @Test(expected = CatalogTransformerException.class)
+    public void testPasswordProtected() throws IOException, CatalogTransformerException {
+        PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+        try (InputStream is = getResource("/password-powerpoint.pptx")) {
+            t.transform(is);
+        }
+    }
+
+    @Test
+    public void testOle2() throws IOException, CatalogTransformerException {
+        PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+        try (InputStream is = getResource("/empty.ppt")) {
+            Metacard mc = t.transform(is);
+            assertEquals(null, mc.getThumbnail());
+        }
+    }
+
+    @Test
+    public void testThumbnailWithEmptySlideShow()
+            throws IOException, CatalogTransformerException, InterruptedException {
+
+        try (XMLSlideShow ss = new XMLSlideShow()) {
+            try (ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+                ss.write(os);
+
+                try (ByteArrayInputStream inStr = new ByteArrayInputStream(os.toByteArray())) {
+                    PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+                    Metacard m = t.transform(inStr);
+                    assertNull(m.getThumbnail());
+                }
+            }
+
+        }
+    }
+
+    @Test
+    public void testWithTitle()
+            throws IOException, CatalogTransformerException, InterruptedException {
+
+        try (XMLSlideShow ss = new XMLSlideShow()) {
+            ss.createSlide();
+            ss.getProperties()
+                    .getCoreProperties()
+                    .setTitle("TheTitle");
+            try (ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+                ss.write(os);
+
+                try (ByteArrayInputStream inStr = new ByteArrayInputStream(os.toByteArray())) {
+                    PptxInputTransformer t = new PptxInputTransformer(inputTransformer);
+                    Metacard m = t.transform(inStr);
+                    assertEquals("TheTitle", m.getTitle());
+                }
+            }
+
+        }
+    }
+
+    private Date createOneSecondPrecisionDate() {
+        return new Date(TimeUnit.SECONDS.toMillis(TimeUnit.MILLISECONDS.toSeconds(new Date().getTime())));
+    }
+
+    @Test
+    public void testCreatedDateWith()
```

Comment:
sorry about that, originally I had a testCreateDateWithout, which made sense at the time :-)